### PR TITLE
Paypro request

### PR DESCRIPTION
**Pull request by matiu**
- send paypro 'payment' message at the moment a TX is broadcasted.
 - parse payment-ack

Closes #2674

### CODE REVIEW

**Comment by matiu** [comment]
updated and ready to be reviews.

File: lib/api.js [hunk 1]
```diff
@@ -1046,10 +1065,31 @@ API.prototype.broadcastTxProposal = function(txp, cb) {
...
-  var url = '/v1/txproposals/' + txp.id + '/broadcast/';
-  self._doPostRequest(url, {}, function(err, txp) {
-    if (err) return cb(err);
-    return cb(null, txp);
+  self.getPayPro(txp, function(err, paypro) {
+
+    if (paypro) {
+
+      var t = WalletUtils.buildTx(txp);
+      self.createAddress(function(err, addr) {
```
Comment (isocolsky) [inline lib/api.js]: Are we planning to switch to a separate branch for refund addresses?

File: test/paypro.js [hunk 2]
```diff
@@ -4,12 +4,12 @@ var _ = require('lodash');
...
 var sinon = sinon || require('sinon');
 var should = chai.should();
-var PayProReq = require('../lib/payprorequest');
+var PayPro = require('../lib/paypro');
 var TestData = require('./testdata');
 
 
-describe('payprorequest', function() {
-  var xhr, http;
+describe('paypro', function() {
```
Comment (isocolsky) [inline test/paypro.js]: I think we need to have a test in place to check that the address we pass as refund address is actually ours. This should be an extra check to the `broadcastTxProposal` function

**Comment by matiu** [comment]
updated with more test. Thanks for the review!

**Comment by isocolsky** [comment]
Great job!

**Pull request merged**